Include/internal/pycore_freelist.h

```diff
@@ -8,27 +8,38 @@ extern "C" {
 #  error "this header requires Py_BUILD_CORE define"
 #endif
 
-#ifndef WITH_FREELISTS
-// without freelists
-#  define PyList_MAXFREELIST 0
-#endif
-
-/* Empty list reuse scheme to save calls to malloc and free */
-#ifndef PyList_MAXFREELIST
+#ifdef WITH_FREELISTS
+// with freelists
 #  define PyList_MAXFREELIST 80
+#  define PyFloat_MAXFREELIST 100
+#else
+#  define PyList_MAXFREELIST 0
+#  define PyFloat_MAXFREELIST 0
 #endif
 
 struct _Py_list_state {
-#if PyList_MAXFREELIST > 0
+#ifdef WITH_FREELISTS
     PyListObject *free_list[PyList_MAXFREELIST];
     int numfree;
 #endif
 };
 
+struct _Py_float_state {
+#ifdef WITH_FREELISTS
+    /* Special free list
+       free_list is a singly-linked list of available PyFloatObjects,
+       linked via abuse of their ob_type members. */
+    int numfree;
+    PyFloatObject *free_list;
+#endif
+};
+
 typedef struct _Py_freelist_state {
+    struct _Py_float_state float_state;
```

Comment:
Why not use the same naming scheme? The list freelist state member is named `list`.

Reply:
Because float is reserved name?

Reply:
If we decide to change the naming schme, list should be changed. WDYT?

Reply:
I think that the naming of a struct and its members is an important part of API design. Having a consistent scheme makes it easier to read and review code. I would suggest use a naming scheme for the freelist struct and its members. Up to you :)

Reply:
Totally agree, I will create a seperate PR :)